runtime/near-vm-runner/src/preload.rs

```diff
@@ -0,0 +1,136 @@
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::Arc;
+
+use threadpool::ThreadPool;
+
+use near_primitives::hash::CryptoHash;
+use near_primitives::runtime::fees::RuntimeFeesConfig;
+use near_primitives::types::CompiledContractCache;
+use near_vm_errors::VMError;
+use near_vm_logic::profile::ProfileData;
+use near_vm_logic::types::PromiseResult;
+use near_vm_logic::{External, ProtocolVersion, VMConfig, VMContext, VMKind, VMOutcome};
+
+use crate::cache;
+use crate::preload::VMModule::Wasmer0;
+use crate::wasmer_runner::run_wasmer_module;
+
+enum VMModule {
+    Wasmer0(wasmer_runtime::Module),
+}
+
+struct VMCallData {
+    result: Result<VMModule, VMError>,
+}
+
+struct CallInner {
+    rx: Receiver<VMCallData>,
+}
+
+// TODO: consider using https://crates.io/crates/scoped_threadpool and not clone the data.
+#[derive(Clone)]
+pub struct ContractCallPrepareRequest {
+    pub code_hash: CryptoHash,
+    pub code: Vec<u8>,
+    pub vm_config: VMConfig,
+    pub cache: Option<Arc<dyn CompiledContractCache>>,
+}
+
+#[derive(Clone)]
+pub struct ContractCallPrepareResult {
+    pub handle: Option<usize>,
+    pub error: Option<VMError>,
+}
```

Comment:
Seems like we can simplify this to just 
```suggestion
pub struct ContractCallPrepareResult {
    rx: Receiver<VMCallData>,
}
```

that is, we don't need `CallInner` and the `prepared` vector, `PrepareResult` itself could carry all the data necessary for resolution.

Reply:
Nope, don't want to expose implementation details (such as VM implementation dependencies) in public API of preloader.

Reply:
But the `rx` can be private, no?